src/model.cpp

```diff
@@ -127,9 +140,87 @@ std::string Model::doSerialisation(FORMATS format) const
 
         repr += serialiseUnits(format);
 
-        // Serialize components of the model, imported components have already been dealt with at this point.
+        // Serialise components of the model, imported components have already been dealt with at this point.
         for (size_t i = 0; i < componentCount(); ++i) {
             repr += getComponent(i)->serialise(format);
+            // Build unique variable equivalence pairs (VariableMap) for connections
+            ComponentPtr component = getComponent(i);
+            for (size_t j = 0; j < component->variableCount(); ++j) {
+                VariablePtr variable = component->getVariable(j);
+                if (variable->equivalentVariableCount() > 0) {
```

Comment:
I'd keep it here, as you're comparing a count to some value - the method doesn't return a boolean. So this way is semantically clearer.